specs/LaunchingRegisteredProtocols.md

```diff
@@ -0,0 +1,283 @@
+}
+
+/// Receives `LaunchingExternalProtocol` events for main frame.
+[uuid(e5fea648-79c9-47aa-8314-f471fe627649), object, pointer_default(unique)] 
+interface ICoreWebView2LaunchingExternalProtocolEventHandler: IUnknown {
+  /// Provides the event args for the corresponding event.
+  HRESULT Invoke(
+      [in] ICoreWebView2* sender,
+      [in] ICoreWebView2LaunchingExternalProtocolEventArgs* args);
+}
+
+/// Receives `LaunchingExternalProtocol` events for non-main frame.
+[uuid(a898c12c-f949-474c-913b-428770cfc177), object, pointer_default(unique)] 
+interface ICoreWebView2FrameLaunchingExternalProtocolEventHandler: IUnknown {
+  /// Provides the event args for the corresponding event.
+  HRESULT Invoke(
+      [in] ICoreWebView2Frame* sender,
+      [in] ICoreWebView2LaunchingExternalProtocolEventArgs* args);
+}
+
+/// Event args for `LaunchingExternalProtocol` event.
+[uuid(fc43b557-9713-4a67-af8d-a76ef3a206e8), object, pointer_default(unique)] 
+interface ICoreWebView2LaunchingExternalProtocolEventArgs: IUnknown {
+  /// The uri of the requested protocol.
+
+  [propget] HRESULT Uri([out, retval] LPWSTR* uri);
+
+  /// The host may set this flag to cancel the protocol launch.  If set to
+  /// `TRUE`, the protocol will not be launched. If cancelled, the
+  /// dialog is not displayed regardless of the `Handled` property.
+
+  [propget] HRESULT Cancel([out, retval] BOOL* cancel);
+
+  /// Sets the `Cancel` property. The default value is `FALSE`.
+
+  [propput] HRESULT Cancel([in] BOOL cancel);
+
+  /// The host may set this flag to `TRUE` to hide the dialog.
+  /// The protocol request will continue as normal if it is not
+  /// cancelled, although there will be no default UI shown. The default
+  /// value is `FALSE` and the default external protocol dialog is shown.
+
+  [propget] HRESULT Handled([out, retval] BOOL* handled);
+
+  /// Sets the `Handled` property. The default value is `FALSE`.
+
+  [propput] HRESULT Handled([in] BOOL handled);
+
+  /// The host may set this flag to `TRUE` to disable the always open checkbox. 
+  /// This will also revoke previous permissions given to this protocol/origin.
+  /// The always open checkbox is displayed only when the 
+  /// launch is being requested from a trustworthy origin and gives the user the option to
+  /// always allow this origin to open links of this type in the associated app.
+  /// See [trustworthy origin]
+  /// (https://w3c.github.io/webappsec-secure-contexts/#potentially-trustworthy-origin) 
+  /// for more information regarding trustworthy origins. See [always open checkbox]
+  /// (https://docs.microsoft.com/en-us/DeployEdge/microsoft-edge-policies#externalprotocoldialogshowalwaysopencheckbox)
+  /// for more information regarding the always open checkbox.
+  /// Permissions will be revoked in the future if this property is set to `TRUE` regardless of the 
+  /// values of the `Cancel` or `Handled` properties. 
+  /// The default value is `FALSE`.
```

Comment:
If the end user checks the always open checkbox, and then goes to launch a URI from the same origin or protocol, what happens with this event? I can imagine two cases:
1. This event is not raised and instead the URI is just launched directly because the end user chose to just always open them
2. This event is still raised, but regardless of the Handled property the default dialog won't be shown. If Cancel is false, and Handled is true or false, the default dialog is not displayed and the URI is launched.

2 is probably preferable because in 1 if the end user chooses to always launch the end dev loses control and never gets this event to change anything in the future.